### PR TITLE
enh(Filter): Allow Host group, Service group and Monitoring server criterions to be used using the name insteas of id

### PR DESCRIPTION
**Pull request by bdauria**
## Type of change

- [x] Patch fixing an issue (non-breaking change)

## Target serie

- [x] 21.10.x (master)

### CODE REVIEW

**Review by Jordan91 — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: www/front_src/src/Resources/Listing/api/endpoint.ts [hunk 1]
```diff
@@ -3,11 +3,11 @@ import { buildListingEndpoint, ListingParameters } from '@centreon/ui';
...
 
 export type ListResourcesProps = {
-  hostGroupIds: Array<number>;
-  monitoringServerIds: Array<number>;
+  hostGroups: Array<number>;
+  monitoringServers: Array<number>;
   onlyWithPerformanceData?: boolean;
   resourceTypes: Array<string>;
-  serviceGroupIds: Array<number>;
+  serviceGroups: Array<number>;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Listing/api/endpoint.ts:10]: Are they not `Array<string>` because we filter them by name

Comment (adr-mo) [reply www/front_src/src/Resources/Listing/api/endpoint.ts:10]: 👍  yep array of strings should be sent

File: www/front_src/src/Resources/testUtils/index.ts [hunk 2]
```diff
@@ -6,13 +6,13 @@ import { Filter } from '../Filter/models';
 import { buildResourcesEndpoint } from '../Listing/api/endpoint';
 
 interface EndpointParams {
-  hostGroupIds?: Array<number>;
+  hostGroups?: Array<number>;
```
Comment (Thebarda) [inline www/front_src/src/Resources/testUtils/index.ts:9]:
```suggestion
  hostGroups?: Array<string>;
```

File: www/front_src/src/Resources/testUtils/index.ts [hunk 3]
```diff
@@ -6,13 +6,13 @@ import { Filter } from '../Filter/models';
...
   limit?: number;
-  monitoringServerIds?: Array<number>;
+  monitoringServers?: Array<number>;
```
Comment (Thebarda) [inline www/front_src/src/Resources/testUtils/index.ts:11]:
```suggestion
  monitoringServers?: Array<string>;
```

File: www/front_src/src/Resources/testUtils/index.ts [hunk 4]
```diff
@@ -6,13 +6,13 @@ import { Filter } from '../Filter/models';
...
   page?: number;
   resourceTypes?: Array<string>;
   search?: string;
-  serviceGroupIds?: Array<number>;
+  serviceGroups?: Array<number>;
```
Comment (Thebarda) [inline www/front_src/src/Resources/testUtils/index.ts:15]:
```suggestion
  serviceGroups?: Array<string>;
```

File: www/front_src/src/Resources/Listing/api/endpoint.ts [hunk 5]
```diff
@@ -3,11 +3,11 @@ import { buildListingEndpoint, ListingParameters } from '@centreon/ui';
 import { resourcesEndpoint } from '../../api/endpoint';
 
 export type ListResourcesProps = {
-  hostGroupIds: Array<number>;
-  monitoringServerIds: Array<number>;
+  hostGroups: Array<number>;
+  monitoringServers: Array<number>;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Listing/api/endpoint.ts:7]:
```suggestion
  hostGroups: Array<string>;
  monitoringServers: Array<string>;
```

File: www/front_src/src/Resources/Listing/api/endpoint.ts [hunk 6]
```diff
@@ -3,11 +3,11 @@ import { buildListingEndpoint, ListingParameters } from '@centreon/ui';
 import { resourcesEndpoint } from '../../api/endpoint';
 
 export type ListResourcesProps = {
-  hostGroupIds: Array<number>;
-  monitoringServerIds: Array<number>;
+  hostGroups: Array<number>;
```
Comment (bdauria) [inline www/front_src/src/Resources/Listing/api/endpoint.ts:6]:
```suggestion
  hostGroups: Array<string>;
```

File: www/front_src/src/Resources/Listing/api/endpoint.ts (hunk 5 above)
Comment (bdauria) [inline www/front_src/src/Resources/Listing/api/endpoint.ts:7]:
```suggestion
  monitoringServers: Array<string>;
```

File: www/front_src/src/Resources/Listing/api/endpoint.ts [hunk 7]
```diff
@@ -3,11 +3,11 @@ import { buildListingEndpoint, ListingParameters } from '@centreon/ui';
...
 
 export type ListResourcesProps = {
-  hostGroupIds: Array<number>;
-  monitoringServerIds: Array<number>;
+  hostGroups: Array<string>;
+  monitoringServers: Array<string>;
   onlyWithPerformanceData?: boolean;
   resourceTypes: Array<string>;
-  serviceGroupIds: Array<number>;
+  serviceGroups: Array<number>;
```
Comment (Thebarda) [inline www/front_src/src/Resources/Listing/api/endpoint.ts:10]:
```suggestion
  serviceGroups: Array<string>;
```

**Review by kduret — DISMISSED** [review]

**Review by Thebarda — DISMISSED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**